misc/dbt-materialize/dbt/include/materialize/macros/utils/generate_cluster_name.sql

```diff
@@ -14,9 +14,11 @@
 -- limitations under the License.
 
 {% macro generate_deploy_cluster_name(custom_cluster_name) -%}
-    {{ custom_cluster_name }}_dbt_deploy
+        {{ custom_cluster_name }}_dbt_deploy
 {%- endmacro %}
 
 {% macro generate_cluster_name(custom_cluster_name) -%}
-    {{ custom_cluster_name }}
+    {% if custom_cluster_name -%}
+        {{ custom_cluster_name }}
+    {%- endif %}
```

Comment:
If no cluster is defined in the profile.yml or as a configuration during a blue green deployment, we would end up with `_dbt_deploy` cluster name. It is better to fail and inform the user about this as we do in other places.

Tests are failing as:

1. In the `generate_final_cluster_name` method, `cluster_name` is initially `None`.

2. When `execute_macro("generate_cluster_name", ...)` is called, it returns an empty string because `custom_cluster_name` is `None`.

3. Then, when `execute_macro("generate_deploy_cluster_name", ...)` is called, it's essentially trying to append `_dbt_deploy` to an empty string, resulting in just `_dbt_deploy`.

```suggestion
{% macro generate_cluster_name(custom_cluster_name) -%}
    {% if var('deploy', false) and not custom_cluster_name -%}
        {{ exceptions.raise_compiler_error("When 'deploy' is set, you must specify a cluster name in your profile or via the cluster config") }}
    {% elif custom_cluster_name -%}
        {{ custom_cluster_name }}
    {%- endif %}
{%- endmacro %}
```

Reply:
For deployments, shouldn't this fail before it reaches `generate_cluster_name`, [here](https://github.com/MaterializeInc/materialize/blob/f880946bb5a89ffcc1f086430e4172bea15e139e/misc/dbt-materialize/dbt/include/materialize/macros/ci/create_cluster.sql#L43)?

Reply:
The difference here is that this works just fine in the deploy actions, we already have a check if a cluster is defined in the deployment vars for those deploy actions. Here we check if a cluster is provided in the dbt run, added more details bellow.